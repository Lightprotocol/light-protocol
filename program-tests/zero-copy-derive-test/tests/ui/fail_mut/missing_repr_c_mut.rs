use light_zero_copy_derive::ZeroCopyMut;

// This should fail because it's missing #[repr(C)]
#[derive(ZeroCopyMut)]
pub struct MissingReprCMut {
    pub field1: u32,
    pub field2: Vec<u8>,
}

fn main() {}