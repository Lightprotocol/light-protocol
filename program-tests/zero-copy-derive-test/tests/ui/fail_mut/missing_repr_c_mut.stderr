error: ZeroCopyMut requires #[repr(C)] attribute for memory layout safety. Add #[repr(C)] above the zerocopymut declaration.
 --> tests/ui/fail_mut/missing_repr_c_mut.rs:4:10
  |
4 | #[derive(ZeroCopyMut)]
  |          ^^^^^^^^^^^
  |
  = note: this error originates in the derive macro `ZeroCopyMut` (in Nightly builds, run with -Z macro-backtrace for more info)
